tests/test_collections.py

```diff
@@ -263,5 +263,96 @@ def test_iteration(self):
         self.assertEqual(m['A','A'], {'energy':3.0, 'mass':2.0})
         self.assertEqual(m['B','B'], {'energy':3.0, 'mass':2.0})
 
+class test_KeyedArray(unittest.TestCase):
```

Comment:
These all look good, but you might need to add a couple more if you want to support both element-wise and scalar arithmetic in each op.